Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -121,4 +121,11 @@ Ref<CSSValue> ViewTimeline::toCSSValue(const RenderStyle& style) const
     );
 }
 
+AnimationTimeline::ShouldUpdateAnimationsAndSendEvents ViewTimeline::documentWillUpdateAnimationsAndSendEvents()
+{
+    if (m_subject)
+        return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::Yes;
+    return AnimationTimeline::ShouldUpdateAnimationsAndSendEvents::No;
+}
+
```

Comment:
This is unnecessary since `ViewTimeline` is a subclass of `ScrollTimeline`.

Reply:
An important thing to note here is the ViewTimelines will not have their m_source set, just their m_subject. We'll need to compute the m_source by looking up the ancestor chain for the nearest scrollable container (like we do now after https://github.com/graouts/WebKit/commit/691b5a8ea25487df174266f4f39cb55763c452fa). So I think this function should remain separate as ViewTimeline needs to check that m_subject exists (and will eventually check its source, once the code to compute it lands).